virttest/qemu_devices/qdevices.py

```diff
@@ -359,6 +359,13 @@ def is_pcie_device(self):
                         "vhost-user-fs-pci"]
         return (driver in pcie_drivers or driver.startswith("virtio-"))
 
+    def _parameters_replace(self):
+        mapping = {"serial": ("NO_EQUAL_STRING", "on")}
```

Comment:
I believe serial here is a specific device's property, ie. not all devices have the serial property, but we call _parameters_replace in all _cmdline_json functions of devices. Besides, we put it in QBaseDevice,  IMO, this class should not handle its children deivces' properties, let the specific device itself handle its property. What's your opinion @nickzhq @YongxueHong @luckyh

Reply:
> I believe serial here is a specific device's property, ie. not all devices have the serial property, but we call _parameters_replace in all _cmdline_json functions of devices. Besides, we put it in QBaseDevice, IMO, this class should not handle its children deivces' properties, let the specific device itself handle its property. What's your opinion @nickzhq @YongxueHong @luckyh

Hi @zhencliu 
Yeah, I agree with you, I also think that the `NO_EQUAL_STRING` is just to handle the serial option and it just represents the options that are without any value then let the framework handle these options without `=` symbol in the command line.
https://github.com/avocado-framework/avocado-vt/blob/247daa8b0bd3b9e718928e46dc35e4ad44fe09ff/virttest/qemu_vm.py#L514
```
                # "NO_EQUAL_STRING": set option as a option string only,
                #                    even without "=".
                #      (In most case, qemu-kvm should recognize it as "<null>")
```
But I don't know how to handle such options in the json command line.